greedy_algo.py

```diff
@@ -72,60 +76,98 @@ def run_replication(self):
   # delta: specify the amount of request increased every time 
   def enough_replica_on_increase(self, delta):
     for c in self.content_set:
-      for a in self.client_set:
-        # add a small amount of requests for content c from client a
-        self.access_map[c][a] += delta
-        # test whether current replicas can handle that much request
-        is_enough = self.enough_replica()
-        # back tracking,
-        self.access_map[c][a] -= delta
-        return is_enough
+      if c in self.access_map:
+        for a in self.access_map[c].keys():
+          # add a small amount of requests for content c from client a
+          self.access_map[c][a] += delta
+          # test whether current replicas can handle that much request
+          is_enough = self.enough_replica()
+          # back tracking,
+          self.access_map[c][a] -= delta
+          if not is_enough:
+            return False
+    return True
 
   def add_replica(self, request_delta, replica_delta):
     I = []
     for c in self.content_set:
```

Comment:
wait... can't we just get the results from `enough_replica_on_increase`? why another for loop here?